src/components/VmDetails/cards/DetailsCard/index.js

```diff
@@ -436,12 +486,29 @@ class DetailsCard extends React.Component {
       vmUpdates['bootMenuEnabled'] = stateVm.get('bootMenuEnabled')
     }
 
+    if (this.trackUpdates['cloudInit']) {
+      vmUpdates['cloudInit'] = stateVm.get('cloudInit').toJS()
+    }
+
+    if (this.trackUpdates['bootDevices']) {
+      vmUpdates['os'] = {
+        bootDevices: stateVm.getIn(['os', 'bootDevices']),
```

Comment:
Needs a `.toJS()`

Could even slice the array to make sure only 2 boot devices are set.

Reply:
Note: So technically having an ImmutableJS object attached in normal object will convert ok with `JSON.stringify(obj)`.  Adding the `toJS()` is nice for devs so we can verify contents of `vmUpdates` without clicking through the object structure in the debugger.